src/Functions/IFunction.cpp

```diff
@@ -266,6 +272,73 @@ ColumnPtr IExecutableFunction::execute(const ColumnsWithTypeAndName & arguments,
     return result;
 }
 
+ColumnPtr IExecutableFunction::execute(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count, bool dry_run) const
+{
+    if (useDefaultImplementationForSparseColumns())
+    {
+        size_t num_sparse_columns = 0;
+        size_t num_full_columns = 0;
+        size_t sparse_column_position = 0;
+
+        for (size_t i = 0; i < arguments.size(); ++i)
+        {
+            const auto * column_sparse = checkAndGetColumn<ColumnSparse>(arguments[i].column.get());
+            /// In rare case, when sparse column doesn't have default values,
+            /// it's more convenient to convert it to full before execution of function.
+            if (column_sparse && column_sparse->getNumberOfDefaults())
+            {
+                sparse_column_position = i;
+                ++num_sparse_columns;
+            }
+            else if (!isColumnConst(*arguments[i].column))
+            {
+                ++num_full_columns;
+            }
+        }
+
+        auto columns_without_sparse = arguments;
+        if (num_sparse_columns == 1 && num_full_columns == 0)
+        {
+            auto & arg_with_sparse = columns_without_sparse[sparse_column_position];
+            ColumnPtr sparse_offsets;
+            {
+                /// New scope to avoid possible mistakes on dangling reference.
+                const auto & column_sparse = assert_cast<const ColumnSparse &>(*arg_with_sparse.column);
+                sparse_offsets = column_sparse.getOffsetsPtr();
+                arg_with_sparse.column = column_sparse.getValuesPtr();
```

Comment:
Note: it is probably possible to get sparse column without default values. But `column_sparse.getValuesPtr()` always contain one. So, something like `intDiv(1 / sparce_column)` could throw when shouldn't.

Reply:
Yes. There is already check for such case a few lines above. If there are no defaults in column, it's converted to full.